### DISCUSSION TITLE
sending CC messages

### DISCUSSION THREAD

**Discussion by Tomartsh** [djipco/webmidi#275] [Q&A] [answered]
Hi,

I may be wrong as I am a newbie, but I did not see an example or documentation concerning sending out CC messages.
I can see them in an event handler though.

Is there such an example ?

Many thanks in advance,
Nino Uziel

**Comment by djipco** [comment, marked as answer]
You are right that there is no such example. However, the documentation for **[Output.sendControlChange()](https://webmidijs.org/api/classes/Output#sendControlChange)** and **[OutputChannel.sendControlChange()](https://webmidijs.org/api/classes/OutputChannel#sendControlChange)** does detail how to do it.

Here's an example that sends a **coarse volume** control change message to all channels of an `Output`:

```js

WebMidi
  .enable()
  .then(()=> {
    const device = WebMidi.getOutputByName("Some device..."); // "device" references an 'Output' object
    device.sendControlChange("volumecoarse", 64);
  });

```
Of course, you need to change "Some device..." by the name of a device available on your system. A list of available devices can be obtained with `console.log(WebMidi.outputs)`

Here's a second example that sends a **chorus level** control change message to MIDI channel 10 of a specific `Output`:

```js

WebMidi
  .enable()
  .then(()=> {
    const device = WebMidi.getOutputByName("Some device...");
    const instrument = device.channels[10]; // "instrument" references an 'OutputChannel' object
    instrument.sendControlChange("choruslevel", 127);
  });

```


Hope this helps!

**Reply by Tomartsh** [reply]
This helps a lot. Managed to make it work.
Many thanks for your prompt assistance
